src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.TaskParseException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * parser to make sense of user commands, parsing from command to action
+ */
+public class Parser {
+    /**
+     * if command is delete, use this method to get the index of task user wants to delete
+     * @param text original command
```

Comment:
Maybe add an empty line between the description and parameter section to follow the Java coding standard?
It is the same for other JavaDoc comments too.